databases/backends/postgres.py

```diff
@@ -140,7 +140,10 @@ def __getitem__(self, key: typing.Any) -> typing.Any:
         elif isinstance(key, int):
             idx, datatype = self._column_map_int[key]
         else:
-            idx, datatype = self._column_map[key]
+            try:
+                idx, datatype = self._column_map[key]
+            except KeyError:
+                return self._row[key]
```

Comment:
Should we be setting `raw` to `self._row[key]` to ensure the processor runs on it still?